modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -111,6 +124,31 @@ public void onParameter(ParameterizedModel owner, ParameterModel model) {
         validateNameCollisionWithTypes(model, ownerName, ownerModelType, extensionModelName,
                                        owner.getParameterModels().stream().map(p -> hyphenize(p.getName())).collect(toList()));
       }
+
+      @Override
+      public void onOperation(HasOperationModels owner, OperationModel model) {
+        model.getParameterModels().forEach(parameterModel -> parameterModel.getModelProperty(ImplementingTypeModelProperty.class)
+            .ifPresent(p -> validateParameterFieldsHaveGetters(model, p.getType())));
+
+        if (model.getOutput() != null) {
+          validateParameterFieldsHaveGetters(model, getType(model.getOutput().getType()));
+        }
+
+        if (model.getOutputAttributes() != null) {
+          validateParameterFieldsHaveGetters(model, getType(model.getOutputAttributes().getType()));
+        }
+      }
+
+      @Override
+      public void onSource(HasSourceModels owner, SourceModel model) {
+        if (model.getOutput() != null) {
```

Comment:
why do we check for null here? `model.getOutput()` should never be null, if a component has no output then it's type should be `VoidType` and that's it.